src/common/sv2_noise.h

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2023-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_COMMON_SV2_NOISE_H
+#define BITCOIN_COMMON_SV2_NOISE_H
+
+#include <compat/compat.h>
+#include <crypto/poly1305.h>
+#include <key.h>
+#include <pubkey.h>
+#include <random.h>
+#include <streams.h>
+#include <uint256.h>
+
+/** The Noise Protocol Framework
+ *  https://noiseprotocol.org/noise.html
+ *  Revision 38, 2018-07-11
+ *
+ *  Stratum v2 handshake and cipher specification:
+ *  https://github.com/stratum-mining/sv2-spec/blob/main/04-Protocol-Security.md
+ */
+
+/** Section 3: All Noise messages are less than or equal to 65535 bytes in length. */
+static constexpr size_t NOISE_MAX_CHUNK_SIZE = 65535;
+/** Sv2 spec 4.5.2 */
```

Comment:
nit: Is that comment needed? SV2 spec 4.5.2 is not relevant to message length.

Reply:
Looks like this was duplicated from the comment below. Dropped.